LLMonFHIR/FHIRInterpretation/UserStudy/UserStudyChatView.swift

```diff
@@ -53,13 +53,27 @@ struct UserStudyChatView: View {
                     ),
                     content: taskInstructionSheet
                 )
+                .viewStateAlert(state: viewModel.llmSession.state)
                 .onAppear(perform: viewModel.startSurvey)
                 .onChange(of: viewModel.llmSession.context, initial: true) {
                     Task {
                         if viewModel.isProcessing {
                             viewModel.updateProcessingState()
                         }
                         _ = await viewModel.generateAssistantResponse()
+                        
+                        // Alerts and sheets can not be displayed at the same time.
+                        if case let .error(error) = viewModel.llmSession.state {
+                            // We have to first dismiss all sheets.
+                            viewModel.setSurveyViewPresented(false)
+                            viewModel.setTaskInstructionSheetPresented(false)
+                            // Wait for animation to complete
+                            try await Task.sleep(for: .seconds(1))
+                            // Re-set the error state.
+                            viewModel.llmSession.state = .generating
```

Comment:
We need to set `isProcessing` false otherwise the chat bubble we be present.

Reply:
Good point; fixed this and improved more of the error handling for subsequent request or if no sheet is presented.